src/history/hs.c

```diff
@@ -1724,29 +1724,21 @@ static int
 __verify_history_store_id(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, uint32_t this_btree_id)
 {
     WT_CURSOR *hs_cursor;
+    WT_CURSOR_BTREE *hs_cbt;
     WT_DECL_ITEM(prev_hs_key);
     WT_DECL_RET;
     WT_ITEM hs_key;
     wt_timestamp_t hs_start_ts;
     uint64_t hs_counter;
     uint32_t btree_id;
     int cmp;
-    bool found;
 
     hs_cursor = session->hs_cursor;
+    hs_cbt = (WT_CURSOR_BTREE *)hs_cursor;
     WT_CLEAR(hs_key);
 
     WT_ERR(__wt_scr_alloc(session, 0, &prev_hs_key));
 
-    /*
```

Comment:
As we are using internal functions to search the data store. We don't need to set the cursor flag anymore.

Reply:
I'd rather leave this flag in. I know it happens to not be necessary now because of how we are (mis)using cursors, but I wouldn't think to add it back in if I changed this code to use a standard cursor, or moved the place where the check happens in the cursor code.